src/psyclone/tests/psyir/transformations/read_only_verify_test.py

```diff
@@ -112,7 +112,7 @@ def test_read_only_options():
                            options={"region_name": ("a", "b")})
     code = str(invoke.gen())
 
-    assert 'CALL read_only_verify_psy_data%PreStart("a", "b", 2, 2)' in code
+    assert 'CALL read_only_verify_psy_data%PreStart("a", "b", 4, 4)' in code
```

Comment:
The first integer is the number of PreVars and has increased by 2 in most examples because now it finds out about the bounds variables (which is what fixes #969).

Just in this example the number of  PostVars (second integer) has also increased. I can see that indeed there are 4 variables provided in the PostStart area, but I am not sure what PostVars are and if the code is working as intended. Maybe @hiker can shed some light. The resulting code looks like this:
```
      DO j=cv_fld%internal%ystart,cv_fld%internal%ystop
        !
        ! ReadOnlyVerifyStart
        !
        CALL read_only_verify_psy_data%PreStart("a", "b", 4, 4)
        CALL read_only_verify_psy_data%PreDeclareVariable("cv_fld%internal%xstart", cv_fld%internal%xstart)
        CALL read_only_verify_psy_data%PreDeclareVariable("cv_fld%internal%xstop", cv_fld%internal%xstop)
        CALL read_only_verify_psy_data%PreDeclareVariable("p_fld", p_fld)
        CALL read_only_verify_psy_data%PreDeclareVariable("v_fld", v_fld)
        CALL read_only_verify_psy_data%PreDeclareVariable("cv_fld%internal%xstart", cv_fld%internal%xstart)
        CALL read_only_verify_psy_data%PreDeclareVariable("cv_fld%internal%xstop", cv_fld%internal%xstop)
        CALL read_only_verify_psy_data%PreDeclareVariable("p_fld", p_fld)
        CALL read_only_verify_psy_data%PreDeclareVariable("v_fld", v_fld)
        CALL read_only_verify_psy_data%PreEndDeclaration
        CALL read_only_verify_psy_data%ProvideVariable("cv_fld%internal%xstart", cv_fld%internal%xstart)
        CALL read_only_verify_psy_data%ProvideVariable("cv_fld%internal%xstop", cv_fld%internal%xstop)
        CALL read_only_verify_psy_data%ProvideVariable("p_fld", p_fld)
        CALL read_only_verify_psy_data%ProvideVariable("v_fld", v_fld)
        CALL read_only_verify_psy_data%PreEnd
        DO i=cv_fld%internal%xstart,cv_fld%internal%xstop
          CALL compute_cv_code(i, j, cv_fld%data, p_fld%data, v_fld%data)
        END DO
        CALL read_only_verify_psy_data%PostStart
        CALL read_only_verify_psy_data%ProvideVariable("cv_fld%internal%xstart", cv_fld%internal%xstart)
        CALL read_only_verify_psy_data%ProvideVariable("cv_fld%internal%xstop", cv_fld%internal%xstop)
        CALL read_only_verify_psy_data%ProvideVariable("p_fld", p_fld)
        CALL read_only_verify_psy_data%ProvideVariable("v_fld", v_fld)
        CALL read_only_verify_psy_data%PostEnd
        !
        ! ReadOnlyVerifyEnd
        !
      END DO
```

Reply:
Hmm, it looks as though it's picking up the loop bounds as being written to and therefore is outputting them *after* the loop as well as before it. As you say, perhaps @hiker can help here?

Reply:
Could this be a structure granularity problem. Since the kernel writes to `cv_fld%data` all the `cv_fld` is a write dependency, and therefore the loop bounds are as well?

Reply:
That might be it. In the past I've implemented code that looked for a read-only field because the Intel compiler would also make a similar mistake. There's no guarantee that an invoke has a read-only field though. Really, we should fix the dependence analysis. I've notified Joerg on Teams about this discussion.

Reply:
Sorry, missed the @. This is the read-only verification test. Since the loop boundaries are read-only, they are provided after the kernel call so it can be verified that they are not changed. And the change from 2 to 4 is indeed that now the loop boundaries are identified.

Reply:
Thanks Joerg!